src/utils.js

```diff
@@ -158,11 +158,14 @@ export function getEdgeOffset(node, parent, offset = {left: 0, top: 0}) {
   if (!node) {
     return undefined;
   }
-
-  // Get the actual offsetTop / offsetLeft value, no matter how deep the node is nested
+  
+  const nodeRect = node.getBoundingClientRect();
+  const parentRect = node.parentNode.getBoundingClientRect();
+  
+  // get distance to parent edges, no matter how deep the node is nested
   const nodeOffset = {
-    left: offset.left + node.offsetLeft,
-    top: offset.top + node.offsetTop,
+    left: offset.left + nodeRect.left - parentRect.left,
+    top: offset.top + nodeRect.top - parentRect.top,
   };
 
   if (node.parentNode === parent) {
```

Comment:
There is another bug here that has been filed and is also affecting my use of this library, as well.  The use of ```node.parentNode``` with a getContainer function can cause a recursive loop that ends with #document.parentNode, which is ```null``` and results in an ```undefined``` return value for offsetEdge.

Changing this to ```node.offsetParent``` resolves this issue and stops the recursion at the scrolling container.

```suggestion
  if (node.offsetParent=== parent) {
```

Reply:
As was mentioned in PR description the problem is using `offset(Top|Left|Parent)`.

So:
- If `parent` is correct ancestor for `node` we will found correct distance using `getBoundingClientRect` recursively.
- If `parent` isn't correct ancestor current implementation will throw exception at `node.parentNode.getBoundingClientRect` where `node.parentNode` is null.

So i suggests check this case at first line of function `if (!node || !node.parentNode) { return undefined; }`. Should it fix your problem?

Reply:
I think the root of my problem maybe instead comes from the fact that this function returns ```undefined``` at all.  The ```animateNodes()``` function in ```SortableContainer/index.js``` uses the offsetEdge.left/right values directly and does not seem to handle the undefined case.

```js
// from animateNodes()
      const sortingOffset = {
        left:
          this.offsetEdge.left + this.translate.x + containerScrollDelta.left,
        top: this.offsetEdge.top + this.translate.y + containerScrollDelta.top,
      };
```

This only happens when I use my own getContainer element, which is a few levels up in the DOM, and not the SortableContainer element.  Using ```offsetParent``` avoids the recursive loop from getting all the way up to #document and ```null```, but even when it does get that far the offset values are still correct, if they were to be returned.

Reply:
I mean `offsetParent` will be node somewhere between `root` (inclusive) and `node`, so recursive `node.parentNode` will reach it anyway. It is not obvious how to handle If `parent` is not ancestor of `node`.

I have bug just because of `offsetParent` and if it works fine in great amount of cases it is still dangerous to use.